mozilla/fxa
# task(settings): Prompt for password before unlinking account

## Because

- If a user unlinks their account, they will need a password set to access the account in the future.

## This pull request

- Directs users to the set password flow if they attempt to unlink an account but have not yet set a password.

## Issue that this pull request solves

Closes: FXA-7934

## Checklist

_Put an `x` in the boxes that apply_


## Screenshots (Optional)

![image](https://github.com/mozilla/fxa/assets/94418270/6d51886e-daf2-4b0c-be7e-816541ddc230)

## Other information (Optional)

To test locally do the following:

- Make sure you have the secrets for google auth setup in the local config.
- Go to http://localhost:3030/?forceExperiment=thirdPartyAuth&forceExperimentGroup=treatment
- If you were previously logged in you may have to click 'Use a different' account
- Sign up with your google account
- On the settings page, click unlink account
- Observe the message indicating that you must first set a password, and a button that will take goes to the set password flow.

_Note, that once the password has been set, the user will need to try unlinking the account again. This felt intuitive, but if it is not, we can handle this with additional messaging or get UX review for the best course of action. It seems like this approach should be sufficient for now to avoid lock out due to a passwordless account. Note that there will be other items around this issue to improve UX and encourage users to set passwords on accounts that were created via third party auth._
